vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/main/java/com/vaadin/flow/component/treegrid/it/TreeGridExpandRecursivelyPage.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2000-2019 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.treegrid.it;
+
+import com.vaadin.flow.component.html.Label;
+import com.vaadin.flow.component.orderedlayout.VerticalLayout;
+import com.vaadin.flow.component.treegrid.TreeGrid;
+import com.vaadin.flow.data.provider.hierarchy.TreeData;
+import com.vaadin.flow.router.Route;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+
+@Route("tree-grid-expand-recursively")
+public class TreeGridExpandRecursivelyPage extends VerticalLayout {
+
+    TreeGrid<String> grid;
+
+    public TreeGridExpandRecursivelyPage() {
+        grid = new TreeGrid<>();
+        grid.setWidth("350px");
+        grid.setMaxHeight("300px");
+
+        grid.addComponentHierarchyColumn(value -> new Label(value))
+                .setHeader("Name").setAutoWidth(true).setFlexGrow(0);
+        grid.addComponentColumn(value -> new Label(new Date().toString()))
+                .setHeader("Datum");
+        grid.addComponentColumn(value -> new Label(new Date().toString()))
+                .setHeader("Datum 2");
+
+        grid.addExpandListener(event -> {
+            Label label = new Label("Expanded");
```

Comment:
Please don't use `Label` element for text. See the description for V10+ Label at https://vaadin.com/api/platform/10.0.19/com/vaadin/flow/component/html/Label.html

Reply:
Will change to Span